lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Mnemonics.hs

```diff
@@ -47,14 +56,87 @@ spec :: forall t. KnownCommand t => SpecWith ()
 spec = do
     it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
         \address obtained with credentials matches JCLI" $ do
-        Stdout stdout <- generateMnemonicsViaCLI @t []
-        let mnemonics = T.words (T.pack stdout)
+        Stdout m <- generateMnemonicsViaCLI @t []
+        let mnemonics = T.words (T.pack m)
         addr  <- mnemonicsToAccountAddress mnemonics
-        xprv  <- walletRewardCredentials @t mnemonics
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m) mempty
+        c `shouldBe` ExitSuccess
+        T.unpack err `shouldContain` "mnemonic"
+        xprv <- getXprv out
         xpub  <- jcliKeyToPublic xprv
         addr' <- jcliAddressAccount xpub
         addr `shouldBe` addr'
 
+    it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
+        \can use two mnemonic sentences" $ do
+        Stdout m1 <- generateMnemonicsViaCLI @t []
+        Stdout m2 <- generateMnemonicsViaCLI @t ["--size", "12"]
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m1) (T.pack m2)
+        c `shouldBe` ExitSuccess
+        T.unpack err `shouldContain` "mnemonic"
+        T.unpack out `shouldContain` "ed25519e_sk"
+
+    it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
+        \cannot use 1st mnemonic > 15" $ do
+        Stdout m1 <- generateMnemonicsViaCLI @t ["--size", "18"]
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m1) mempty
+        c `shouldBe` ExitFailure 1
+        T.unpack err `shouldContain` "Invalid number of words: 15 words are expected."
+        T.unpack out `shouldContain` mempty
+
+    it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
+        \cannot use 1st mnemonic < 15" $ do
+        Stdout m1 <- generateMnemonicsViaCLI @t ["--size", "12"]
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m1) mempty
+        c `shouldBe` ExitFailure 1
+        T.unpack err `shouldContain` "Invalid number of words: 15 words are expected."
+        T.unpack out `shouldContain` mempty
+
+    it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
+        \cannot use 2nd mnemonic < 9" $ do
+        Stdout m1 <- generateMnemonicsViaCLI @t []
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m1) (T.unwords mnemonics6)
+        c `shouldBe` ExitFailure 1
+        T.unpack err `shouldContain` "Invalid number of words: 9 or 12 words are expected."
+        T.unpack out `shouldContain` mempty
+
+    it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
+        \cannot use 2nd mnemonic > 12" $ do
+        Stdout m1 <- generateMnemonicsViaCLI @t []
+        Stdout m2 <- generateMnemonicsViaCLI @t ["--size", "18"]
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m1) (T.pack m2)
+        c `shouldBe` ExitFailure 1
+        T.unpack err `shouldContain` "Invalid number of words: 9 or 12 words are expected."
+        T.unpack out `shouldContain` mempty
```

Comment:
I don't know why those cases fail. They seem to work fine _manually_...
`cannot use 2nd mnemonic < 9`
`cannot use 2nd mnemonic > 12`

Reply:
@piotr-iohk what do you mean? What does fail exactly?

Reply:
_ok I see now when running the tests_